online-shop/src/app/components/containers/product-details/product-details.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import {
+  getProductDetail,
+  getProductsDetail,
+} from 'src/app/mocks/products.mocks';
+import { ProductDetail } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss'],
+})
+export class ProductDetailsComponent implements OnInit {
+  productDetail: ProductDetail | undefined;
+  constructor(private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.getProductsDetails();
+  }
+  nullUndefined(text: string | null) {
+    if (text == null) return undefined;
+    return text;
+  }
```

Comment:
This function doesn't really make sense, since the Object would be undefined by default, according to the definition above

Reply:
done, I have used it before because I didn't know how to solve an error.